test/Microsoft.AspNetCore.Mvc.Core.Test/Internal/DefaultAssemblyPartDiscoveryProviderTest.cs

```diff
@@ -33,10 +33,10 @@ public void CandidateResolver_ThrowsDifferentCaseAssemblyReference()
                 Enumerable.Empty<RuntimeFallbacks>());
 
             // Act
-            Exception exception = Assert.Throws<Exception>(() => DefaultAssemblyPartDiscoveryProvider.GetCandidateLibraries(dependencyContext));
+            Exception exception = Assert.Throws<InvalidOperationException>(() => DefaultAssemblyPartDiscoveryProvider.GetCandidateLibraries(dependencyContext));
```

Comment:
Could we change the test name `CandidateResolver_ThrowsIfDependencyContextContainsDuplicateRuntimeLibraryNames` or something like that?